flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-hudi/src/main/java/org/apache/flink/cdc/connectors/hudi/sink/coordinator/MultiTableStreamWriteOperatorCoordinator.java

```diff
@@ -0,0 +1,973 @@
+
+                CompactionUtil.scheduleCompaction(
+                        tableContext.writeClient,
+                        state.isDeltaTimeCompaction,
+                        true); // committed = true since we just committed
+
+                // Reset metrics after scheduling
+                state.resetCompactionMetrics();
+
+                LOG.info("Successfully scheduled compaction for table [{}]", tableId);
+            } catch (Exception e) {
+                LOG.error("Failed to schedule compaction for table [{}]", tableId, e);
+                // Don't fail the job, just log the error
+            }
+        } else if (state.scheduleCompaction) {
+            LOG.debug(
+                    "Compaction not triggered for table [{}] - commits={}/{}, bytes={} MB",
+                    tableId,
+                    state.commitsSinceLastCompaction,
+                    state.commitsThreshold,
+                    state.totalLogBytesWritten / (1024 * 1024));
+        }
+
+        // Clustering can remain on every commit or use similar metrics
+        if (state.scheduleClustering) {
+            try {
+                Configuration tableConfig = createTableSpecificConfig(tableId);
+                ClusteringUtil.scheduleClustering(
+                        tableConfig,
+                        tableContext.writeClient,
+                        true); // committed = true since we just committed
+                LOG.info("Scheduled clustering for table [{}]", tableId);
+            } catch (Exception e) {
+                LOG.error("Failed to schedule clustering for table [{}]", tableId, e);
+                // Don't fail the job, just log the error
+            }
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (instantRequestExecutor != null) {
+            instantRequestExecutor.close();
+        }
+        tableContexts.values().forEach(TableContext::close);
+        tableContexts.clear();
+        pathToTableId.clear();
+        super.close();
+        LOG.info("MultiTableStreamWriteOperatorCoordinator closed.");
+    }
+
+    @Override
+    public void subtaskReady(int i, SubtaskGateway subtaskGateway) {
+        // Since the parent's `gateways` field is private, we must manage our own.
+        if (this.gateways == null) {
+            this.gateways = new SubtaskGateway[context.currentParallelism()];
+        }
+        this.gateways[i] = subtaskGateway;
+    }
+
+    // --- Helper Methods ---
+
+    /**
+     * Creates the physical directory for a Hudi table if it doesn't exist. This must be done on the
+     * coordinator side to avoid race conditions when multiple task managers try to create the same
+     * directory simultaneously.
+     *
+     * @param config The table-specific configuration containing the path
+     * @throws IOException if directory creation fails
+     */
+    private static void createHudiTablePath(Configuration config) throws IOException {
```

Comment:
FlinkOptions.PATH here is not table path, but something like db path? can we rename the method to avoid ambiguation?